databuilder/databuilder/task/neo4j_staleness_removal_task.py

```diff
@@ -182,75 +210,84 @@ def _batch_delete(self,
+            stale_records = self._execute_cypher_query(
+                statement=stale_nodes_statement.format(type=target_type,
+                                                       extra_condition=stale_extra_condition),
+                param_dict={MARKER_VAR_NAME: self.marker})
+
+            total_record_value = next(iter(total_records), None)
+            stale_record_value = next(iter(stale_records), None)
+            self._validate_staleness_pct(
+                total_record_count=total_record_value['count'] if total_record_value is not None else 0,
+                stale_record_count=stale_record_value['count'] if stale_record_value is not None else 0,
+                target_type=target_type)
 
     def _validate_relation_staleness_pct(self) -> None:
-        total_relations_statement = textwrap.dedent("""
-        MATCH ()-[r]-()
-        RETURN type(r) as type, count(*) as count;
-        """)
-
-        stale_relations_statement = textwrap.dedent("""
-        MATCH ()-[n]-()
-        WHERE {}
-        RETURN type(n) as type, count(*) as count
-        """)
-
-        stale_relations_statement = textwrap.dedent(self._decorate_staleness(stale_relations_statement))
-
-        total_records = self._execute_cypher_query(statement=total_relations_statement)
-        stale_records = self._execute_cypher_query(statement=stale_relations_statement,
-                                                   param_dict={MARKER_VAR_NAME: self.marker})
-        self._validate_staleness_pct(total_records=total_records,
-                                     stale_records=stale_records,
-                                     types=self.target_relations)
+        total_relations_statement = textwrap.dedent(
+            self.validate_relation_staleness_statement.format(staleness_condition=''))
+        stale_relations_statement = textwrap.dedent(
+            self._decorate_staleness(self.validate_relation_staleness_statement))
+
+        for t in self.target_relations:
+            if isinstance(t, TargetWithCondition):
+                target_type = t.target_type
+                total_extra_condition = 'WHERE ' + t.condition
+                stale_extra_condition = ' AND ' + t.condition
+            else:
+                target_type = t
+                total_extra_condition = ''
+                stale_extra_condition = ''
+
+            total_records = self._execute_cypher_query(
+                statement=total_relations_statement.format(type=target_type,
+                                                           extra_condition=total_extra_condition))
+            stale_records = self._execute_cypher_query(
+                statement=stale_relations_statement.format(type=target_type,
+                                                           extra_condition=stale_extra_condition),
+                param_dict={MARKER_VAR_NAME: self.marker})
+
+            total_record_value = next(iter(total_records), None)
```

Comment:
if you just return the count from the statement instead of the tuple, i believe you won't have to do this weird thing.

Reply:
The `_execute_cypher_query` function returns a `Iterable[Dict[str, Any]]` type which is the reason for this, the same next iter thing is being done in `_batch_delete` when only a count is returned (although I didn't realize it when I wrote this part)